### PR TITLE
Add preferBlockStatements rule

### PR DESCRIPTION
**Pull request by ooflorent**
This new lint rule enforces `BlockStatement` node over single statement nodes.

### CODE REVIEW

**Comment by ooflorent** [comment]
This rule could be improve when it encounters an `EmptyStatement` node by replacing it by a comment:

```js
// before
while (x = x.next);

// after
while (x = x.next) {
  // empty
}
```

However, I did not find how to add a comment...
@sebmck Any suggestion?

File: packages/@romejs/js-compiler/lint/rules/preferBlockStatements.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {CompilerContext, Path} from '@romejs/js-compiler';
+import {AnyNode, AnyStatement, blockStatement} from '@romejs/js-ast';
+import {descriptions} from '@romejs/diagnostics';
+
+function addDiagnostic(context: CompilerContext, statement: AnyStatement): void {
+  context.addFixableDiagnostic(
```
Comment (sebmck) [inline packages/@romejs/js-compiler/lint/rules/preferBlockStatements.ts:13]: Without returning this from `enter` the autofix doesn't work.

**Comment by sebmck** [comment]
To insert a comment you can use the `commentInjector` hook. Example of that is in [`packages/@romejs/js-compiler/lint/suppressions.ts`](https://github.com/facebookexperimental/rome/blob/master/packages/@romejs/js-compiler/lint/suppressions.ts#L67-L73):

```
const id = path.callHook(
  commentInjector,
  {
    type: 'CommentLine',
    value: ` empty`,
  },
);

({innerComments: [id]});
```

File: packages/@romejs/codec-semver/parse.ts [hunk 2]
```diff
@@ -121,7 +121,9 @@ const createSemverParser = createParser((ParserCore) =>
 
     // Remove all subsequent space tokens
     eatSpaceToken() {
-      while (this.eatToken('Space') !== undefined);
+      while (this.eatToken('Space') !== undefined) {
+        // empty
+      }
```
Comment (ooflorent) [inline packages/@romejs/codec-semver/parse.ts:126]: 🎉

File: packages/@romejs/diagnostics/descriptions.ts [hunk 3]
```diff
@@ -312,6 +312,10 @@ export const descriptions = createMessages({
         },
       ],
     }),
+    PREFER_BLOCK_STATEMENT: {
+      category: 'lint/preferBlockStatements',
+      message: 'TODO',
```
Comment (ooflorent) [inline packages/@romejs/diagnostics/descriptions.ts:317]: Suggestions welcomed

Comment (sebmck) [inline packages/@romejs/diagnostics/descriptions.ts:317]: Something simple like

```suggestion
      message: 'Block statements are preferred in this position',
```

Would probably do for now. A lot of the lint messages are pretty bad and shiuld be cleaned up prior to a release. It's easy for us to go over all the diagnostic messages now with this file.

File: packages/@romejs/js-compiler/lint/rules/preferBlockStatements.test.md [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+    for (;;);
+    ^^^^^^^^^ 
+
+  ℹ Recommended fix
+
+      │ - for (;;);
+    1 │ + for (;;)·{
+    2 │ + 
+    3 │ + }
```
Comment (ooflorent) [inline packages/@romejs/js-compiler/lint/rules/preferBlockStatements.test.md:129]: The comment is not shown. This is probably a bug in the reporter.

**Comment by sebmck** [comment]
Looks good! The `dedent` tagged template function will be handy in a lot of other places too.

**Comment by ooflorent** [comment]
I added this helper because it produces better snapshot diff since the indentation issues are no longer present.

**Pull request merged**